src/Yaapii.Atoms/Map/Joined.cs

```diff
@@ -64,24 +64,26 @@ public sealed class Joined : MapEnvelope
         public Joined(IEnumerable<IDictionary<string, string>> dicts, bool rejectBuildingAllValues = true) : base(
             () =>
                 new LazyDict(
-                new Enumerable.Joined<IKvp>(
-                    new Mapped<IDictionary<string, string>, IEnumerable<IKvp>>(dict =>
-                        new ManyOf<IKvp>(
-                            new ScalarOf<IEnumerator<IKvp>>(() =>
-                            {
-                                IEnumerable<IKvp> list = new ManyOf<IKvp>();
-                                foreach (var key in dict.Keys)
+                    new Enumerable.Joined<IKvp>(
+                        new Mapped<IDictionary<string, string>, IEnumerable<IKvp>>(dict =>
+                            new LiveMany<IKvp>(() =>
+                                new Live<IEnumerator<IKvp>>(() =>
                                 {
-                                    list = new Enumerable.Joined<IKvp>(list, new KvpOf(key, () => dict[key]));
-                                }
-                                return list.GetEnumerator();
-                            })
+                                    IEnumerable<IKvp> list = new ManyOf<IKvp>();
+                                    foreach (var key in dict.Keys)
+                                    {
+                                        list = new Enumerable.Joined<IKvp>(list, new KvpOf(key, () => dict[key]));
```

Comment:
I do not like this as still means, that every dict entry would be mapped into a joined decorator...

Imagine joining 2 dicts with 100.000 entries each (which is not a rare thing). The Joined-decorations would stack up accordingly and would probably result in a stack-overflow as described in the issues.
I already made a PR targeting this class specifically: https://github.com/icarus-consulting/Yaapii.Atoms/pull/542

Reply:
You might want to have a look at [this suggestion](https://github.com/icarus-consulting/Yaapii.Atoms/issues/562)